### DISCUSSION TITLE
Provide Role Based Access Control feature to admin API and admin dashboard UI

### DISCUSSION THREAD

**Discussion by avsilva** [medusajs/medusa#5059] [Feature Requests]
It would be nice to have role based access control on the medusa API endpoints. 
The user role could also be used to filter out admin UI components, like links, buttons, etc

**Comment by ghost** [comment]
Any updates on this idea? It looks very impressive.

**Reply by lexvz14** [reply]
https://docs.medusajs.com/modules/users/backend/rbac
There are options to implement roles and permissions based on these roles. I might take it for a spin shortly.

**Reply by ambigos1** [reply]
> https://docs.medusajs.com/modules/users/backend/rbac There are options to implement roles and permissions based on these roles. I might take it for a spin shortly.

The link is broken

**Reply by MattEzekiel** [reply]
@skdishansachin manteiners says that there are no plans for user roles

**Comment by ambigos1** [comment]
I can't believe that a basic and important feature such as roll-based access is not implemented in Medusa.

**Comment by radoslaw-sz** [comment]
My company tried to not fork the Medusa and create a plugin. Here are the results:


https://github.com/user-attachments/assets/27280794-3083-4ee2-bc81-33974c21013b


However, we are still waiting for Medusa v2 to support plugins out of the box as this one won't be open-sourced unfortunately.

**Reply by francisNeetha** [reply]
Hi, I've been struggling with the same issue and I'm really interested in solving it the way you've done here. I'm trying to create a plugin and would love to follow the same approach you've taken. Could you please guide me through the steps or share more details on how you implemented it? If you’ve created any repository related to this, it would be greatly appreciated if you could share it with me.

**Reply by radoslaw-sz** [reply]
hi @francisNeetha  - due to commercial nature of this plugin, I cannot unfortunately share the details... Only via commercial cooperation.

**Reply by francisNeetha** [reply]
Thank you for your response. I Appreciate you getting back to me.

**Comment by codercody** [comment]
Isn't RBAC outlined in the [auth module](https://docs.medusajs.com/resources/commerce-modules/auth)? What RBAC features would you want that aren't outlined here?

**Reply by SalahAdDin** [reply]
not and answer yet

**Comment by guoanjie1994-lang** [comment]
您好！medusa 作为电商平台很灵活，特别欣赏它对 Agent 友好的设计。电商场景（智能客服、商品描述生成、个性化推荐）对 LLM 调用量大且成本敏感。

我们是 OpenAI 兼容的中文模型聚合平台（[https://arkg.com.cn，一个](https://arkg.com.xn--cn,-s18doh/) key 调 DeepSeek V4/GLM/Kimi/Qwen，$0.2/M 起，人民币直充，明码标价），很适合给你们的商家/开发者用户提供一个低成本模型上游。

如果你们在考虑为用户接入模型或自身降本，可以聊聊——我们能提供接口和测试额度，也欢迎交流对接经验。谢谢！
